### PR TITLE
Allow to show response headers in json output without using `-irh` option

### PR DESCRIPTION
**Pull request by JoshuaMart**
Hi,
This PR add the `-irh` / `-include-response-headers` option to display the response headers without having to use `-irr` 

This option allow to reduces the output size as `-irr` also includes the response body, which can considerably increase the size of the JSON if many targets are scanned.

I'm also taking advantage of this PR to pluralize some variables concerning headers.

### CODE REVIEW

File: runner/runner.go [hunk 1]
```diff
@@ -1519,12 +1520,16 @@ retry:
...
 
 	var (
-		serverResponseRaw string
-		request           string
-		rawResponseHeader string
-		responseHeader    map[string]interface{}
+		serverResponseRaw  string
+		request            string
+		rawResponseHeaders string
+		responseHeader     map[string]interface{}
```
Comment (dogancanbakir) [inline runner/runner.go:1526]: responseHeaders*

**Review by dogancanbakir — APPROVED** [review]
LGTM
```console
$ go run . -u scanme.sh -json -irr | jq .

    __    __  __       _  __
   / /_  / /_/ /_____ | |/ /
  / __ \/ __/ __/ __ \|   /
 / / / / /_/ /_/ /_/ /   |
/_/ /_/\__/\__/ .___/_/|_|
             /_/

                projectdiscovery.io

[INF] Current httpx version v1.3.4 (latest)
{
  "timestamp": "2023-09-01T11:26:14.959831968Z",
  "hash": {
    "body_md5": "444bcb3a3fcf8389296c49467f27e1d6",
    "body_mmh3": "-2088429648",
    "body_sha256": "2689367b205c16ce32ed4200942b8b8b1e262dfc70d9bc9fbc77c49699a4f1df",
    "body_simhash": "590632390773907579",
    "header_md5": "25ede3d2a4612ac02e878f3ba11ad522",
    "header_mmh3": "468786967",
    "header_sha256": "33284418b1b002da19d1d0629b475ccb7adcbfc92a93c196a50d6a2e768580bc",
    "header_simhash": "9814386056708733823"
  },
  "port": "443",
  "url": "https://scanme.sh",
  "input": "scanme.sh",
  "scheme": "https",
  "body": "ok",
  "content_type": "text/plain",
  "method": "GET",
  "host": "128.199.158.128",
  "path": "/",
  "headers": {
    "content_length": "2",
    "content_type": "text/plain; charset=utf-8",
    "date": "Fri, 01 Sep 2023 11:26:15 GMT"
  },
  "raw_header": "HTTP/1.1 200 OK\r\nConnection: close\r\nContent-Length: 2\r\nContent-Type: text/plain; charset=utf-8\r\nDate: Fri, 01 Sep 2023 11:26:15 GMT\r\n\r\n",
  "request": "GET / HTTP/1.1\r\nHost: scanme.sh\r\nUser-Agent: Mozilla/5.0 (X11; Linux i686) AppleWebKit/534.34 (KHTML, like Gecko) QupZilla/1.2.0 Safari/534.34\r\nAccept-Charset: utf-8\r\nAccept-Encoding: gzip\r\n\r\n",
  "time": "1.008596584s",
  "a": [
    "128.199.158.128",
    "2400:6180:0:d0::91:1001"
  ],
  "words": 1,
  "lines": 1,
  "status_code": 200,
  "content_length": 2,
  "failed": false,
  "knowledgebase": {
    "PageType": "nonerror"
  }
}
```

File: runner/types.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ type Result struct {
 	FavIconMMH3        string                 `json:"favicon,omitempty" csv:"favicon"`
 	FaviconPath        string                 `json:"favicon_path,omitempty" csv:"favicon_path"`
 	FinalURL           string                 `json:"final_url,omitempty" csv:"final_url"`
-	ResponseHeader     map[string]interface{} `json:"header,omitempty" csv:"header"`
+	ResponseHeaders    map[string]interface{} `json:"headers,omitempty" csv:"headers"`
 	RawHeader          string                 `json:"raw_header,omitempty" csv:"raw_header"`
```
Comment (dogancanbakir) [inline runner/types.go:56]: raw_headers*

Comment (JoshuaMart) [reply runner/types.go:56]: Sorry, I'm not sure to understand what you expect here ? :)

**Review by Mzack9999 — APPROVED** [review]

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**